### PR TITLE
ref: Split symbolication module into smaller pieces

### PR DESCRIPTION
**Pull request by Swatinem**
This splits the main `symbolication/mod.rs` into smaller modules,
in particular this moves all the apple crash report, and symbolication
specific parts out of the main module which now concerns itself with
request dispatching.
However, each of the modules still includes way too much timeout and metrics-related code/wrappers. We might want to clean those up later as well.

These modules could also use a few more comments explaining the separation.

#skip-changelog

### CODE REVIEW

File: crates/symbolicator/src/services/symbolication/mod.rs [hunk 1]
```diff
@@ -6,78 +6,32 @@ use std::sync::atomic::{AtomicUsize, Ordering};
...
-use crate::utils::hex::HexValue;
+use crate::utils::futures::CallOnDrop;
 
+mod apple;
 mod module_lookup;
 mod process_minidump;
+// we should really rename this here to the `SymbolicatorService`, as it does a lot more
+// than just symbolication ;-)
+#[allow(clippy::module_inception)]
+mod symbolication;
```
[the remark this answers is not in the document]
Comment (Swatinem) [reply crates/symbolicator/src/services/symbolication/mod.rs:32]: I’m very open to suggestions ;-) Naming things is one of the hard things

[the remark this answers is not in the document]
Comment (flub) [reply crates/symbolicator/src/services/symbolication/mod.rs:32]: `symbolicate` is maybe a little better but not a lot i guess.

**Review by flub — APPROVED** [review]
nice, smaller files are good.

(though spreading `impl`s over multiple files is still somewhat weird to me)

**Comment by Swatinem** [comment]
The old implementation had all these seperate impl blocks, but still in the same file. That was also very confusing :-D

**Comment by flub** [comment]
yes, that's entirely true!  i'm not letting this stand in the way of improvement :)

**Comment by Swatinem** [comment]
I will rename this in a followup, as otherwise it would cause too many merge conflicts with https://github.com/getsentry/symbolicator/pull/838

**Pull request merged**